consensus/misc/rewards.go

```diff
@@ -42,3 +42,34 @@ func CalculateReward() *big.Int {
 		return nil
 	}
 }
+
+func CalculateRewardWithIndex(index int) *big.Int {
```

Comment:
Rather than create a new function to pass in a context (called index?), please just modify `CalculateReward()` with variatic args, same as everywhere else. Don't want duplicate logic